rotkehlchen/data_import/importers/binance.py

```diff
@@ -499,9 +677,15 @@ def _group_binance_rows(
                 formatstr=timestamp_format,
                 location='binance',
             )
-            csv_row['Coin'] = asset_from_binance(csv_row['Coin'])
-            csv_row['Change'] = deserialize_asset_amount(csv_row['Change'])
-            multirows[timestamp].append(csv_row)
+            multirows[timestamp].append(BinanceCsvRow(  # maintain insertion order for consistent hash  # noqa: E501
```

Comment:
Hmmm that's not what I had in mind when I mentioned TypedDict.

It's true you would need to do something like that to turn an existing dict into typeddict, and probably can't go around it here.

In these cases probably a simple dataclass. And I would do all typing possible here. In this function. Not let it for later.

So I would also save serialized all possible serialized attributes. Like `UTC_Time', 'Account' etc.

But .... do you need those rows? It seems you don't. For example check above. The only part where `UTC_Time` is used is in like 676 right above here. Nowhere else. Then no need to add it here. Same with `User_ID`, 'Remark'. And maybe more?

If nobody reads them do not have them in the dataclass.

Reply:
I can create the data class, I should probably hash the row and store immediately to the dataclass field `hashed`, that’s the only reason I converted the entire row

Reply:
No, why wuld you do that? You can just add a dataclass method to generate the event identifer which does that for you.

Reply:
I made the other fixes, when trying to move this to a dataclass, I realized it would be more messy, unless I did not understand you correctly.

- currently all the fields are accessed when we call `hash_csv_row`
- if I move it to a dataclass, i still have to define all the fields since I need them for the hash, and when serializing the dataclass, I'll be creating a dictionary matching the default out put from dataclass, just to hash it too